sklearn/naive_bayes.py

```diff
@@ -351,9 +351,10 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
         unique_y_in_classes = in1d(unique_y, classes)
 
         if not np.all(unique_y_in_classes):
+            not_unique_y_in_classes = ~unique_y_in_classes
             raise ValueError("The target label(s) %s in y do not exist in the "
                              "initial classes %s" %
-                             (y[~unique_y_in_classes], classes))
+                             (y[np.where(not_unique_y_in_classes)], classes))
```

Comment:
the problem with the boolean array was that it is of the wrong shape, because y is of length ``n_samples`` and ``unique_y_in_classes`` is of length ``n_classes``.
I'm not sure why you think you change would fix anything.